types/sqs-consumer/index.d.ts

```diff
@@ -1,29 +1,34 @@
-// Type definitions for sqs-consumer 3.5
+// Type definitions for sqs-consumer 3.7
 // Project: https://github.com/BBC/sqs-consumer
-// Definitions by: Daniel Chao <http://dchao.co/>, Eric Byers <http://www.github.com/EricByers>
+// Definitions by: Daniel Chao <http://dchao.co/>
+//                 Eric Byers <http://github.com/EricByers>
+//                 Ezinwa Okpoechi <http://github.com/BrainMaestro>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 import { SQS } from "aws-sdk";
 
-export type ConsumerDone = (error?: Error) => void;
+export type ConsumerDone = (error?: SQSError) => void;
 
 export interface Options {
-  queueUrl: string;
-  handleMessage(message: SQS.Message, done: ConsumerDone): any;
-  region?: string;
-  attributeNames?: string[];
-  messageAttributeNames?: string[];
-  batchSize?: number;
-  visibilityTimeout?: number;
-  waitTimeSeconds?: number;
-  authenticationErrorTimeout?: number;
-  sqs?: SQS;
+    queueUrl: string;
+    handleMessage(message: SQS.Message, done: ConsumerDone): any;
+    region?: string;
+    attributeNames?: string[];
+    messageAttributeNames?: string[];
+    batchSize?: number;
+    visibilityTimeout?: number;
+    waitTimeSeconds?: number;
+    authenticationErrorTimeout?: number;
+    sqs?: SQS;
 }
 
-export interface Consumer extends NodeJS.EventEmitter {
-  start(): void;
-  stop(): void;
+declare class SQSError extends Error {}
+
+declare class Consumer extends NodeJS.EventEmitter {
```

Comment:
If we're declaring this as a class, we should provide correct typings for the constructor as well.

The source code annotates the options that the constructor accepts:

https://github.com/bbc/sqs-consumer/blob/master/index.js#L52